src/jingle_sip/sip_to_jingle.erl

```diff
@@ -0,0 +1,300 @@
+sdp_group_content_to_el(Content) ->
+    #xmlel{name = <<"content">>,
+           attrs = [{<<"name">>, Content}]}.
+
+
+%% This function assumes that all codecs and their attributes were
+%% extracted from the attributes list.
+%% There are still attributes in the list describing other content parameters
+sdp_media_to_content_el(#sdp_m{media = Media, attributes = Attrs}, CodecMap) ->
+    Codecs = maps:get(Media, CodecMap),
+    PayloadEls = [codec_to_payload(Codec) || Codec <- Codecs],
+    Content  = parse_nksip_media_attrs(Attrs),
+    AdditionalDescriptionEls = description_elements_from_content(Content),
+
+    DescriptionEl = #xmlel{name = <<"description">>,
+                           attrs = [{<<"xmlns">>, <<"urn:xmpp:jingle:apps:rtp:1">>},
+                                    {<<"media">>, Media}],
+                           children = PayloadEls ++ AdditionalDescriptionEls},
+    TransportEl = make_transport_el(maps:get(transport, Content)),
+
+    #xmlel{name = <<"content">>,
+           attrs = [{<<"creator">>, <<"initiator">>},
+                    {<<"name">>, maps:get(name, Content, Media)},
+                    {<<"senders">>, maps:get(sender, Content)}],
+           children = [DescriptionEl, TransportEl]}.
+
+codec_to_payload({ID, RTPMap, Attrs}) ->
+    Parameters = lists:flatmap(fun codec_attr_to_params/1, Attrs),
+    {Name, ClockRate, Channels} = parse_codec_rtpmap(RTPMap),
+    #xmlel{name = <<"payload-type">>,
+           attrs = [{<<"id">>, ID},
+                    {<<"name">>, Name},
+                    {<<"clockrate">>, ClockRate},
+                    {<<"channels">>, Channels}],
+          children = Parameters}.
+
+parse_codec_rtpmap(RTPMap) ->
+    case binary:split(RTPMap, <<$/>>, [global]) of
+        [Name, ClockRate] ->
+            {Name, ClockRate, <<"1">>};
+        [Name, ClockRate, Channels] ->
+            {Name, ClockRate, Channels}
+    end.
+
+codec_attr_to_params({<<"rtcp-fb">>, Value}) ->
+    Attrs = rtcp_fb_value_to_xml_attrs(Value),
+    [#xmlel{name = <<"rtcp-fb">>,
+            attrs = [{<<"xmlns">>, <<"urn:xmpp:jingle:apps:rtp:rtcp-fb:0">>} | Attrs]}];
+codec_attr_to_params({<<"fmtp">>, [Value]}) ->
+    Params = binary:split(Value, <<$;>>, [global]),
+    [basic_param_to_xml_el(Param) || Param <- Params];
+codec_attr_to_params(_) ->
+    [].
+
+rtcp_fb_value_to_xml_attrs([Type]) ->
+    [{<<"type">>, Type}];
+rtcp_fb_value_to_xml_attrs([Type, SubType]) ->
+    [{<<"type">>, Type}, {<<"subtype">>, SubType}].
+
+basic_param_to_xml_el(Param) ->
+    case binary:split(Param, <<$=>>) of
+        [Name, Value] ->
+            #xmlel{name = <<"parameter">>,
+                   attrs = [{<<"name">>, Name},
+                            {<<"value">>, Value}]};
+        [Value] ->
+            #xmlel{name = <<"parameter">>,
+                   attrs = [{<<"value">>, Value}]}
+    end.
+
+parse_nksip_media_attrs(Attrs) ->
+    parse_nksip_media_attrs(Attrs, #{}).
+
+parse_nksip_media_attrs([], Acc) ->
```

Comment:
Can we try

```erlang
+parse_nksip_media_attrs([], Acc) ->
+    Acc;
+parse_nksip_media_attrs([{AttrName, AttrValue} | Rest], Acc) ->
+    NewAcc = parse_nksip_media_attr(AttrName, AttrValue, Acc),
+    parse_nksip_media_attrs(Rest, NewAcc).
```

instead?

Than parse_nksip_media_attr/3 would be more readable

```erlang
+parse_nksip_media_attr(<<"sendrecv">>, _, Acc) ->
+    Acc#{sender => <<"both">>};
+parse_nksip_media_attr(<<"recvonly">>, _, Acc) ->
+    Acc#{sender => <<"responder">>};
+parse_nksip_media_attr(<<"sendonly">>, _, Acc) ->
+    Acc#{sender => <<"initiator">>};
```

Reply:
Done.